crypto4pl.c

```diff
@@ -66,6 +68,255 @@ typedef enum
 } crypt_mode_t;
 
 
+                 /***************************
+                 *         HASHING          *
+                 ****************************/
+
+
+#define CONTEXT_MAGIC (~ 0x53481284L)
+
+typedef struct context
+{ int             magic;
+  atom_t          atom;
+  unsigned int    encoding;
```

Comment:
I used `md54pl.c` as an implementation example. Changed in the latest update of this pull request.